### PR TITLE
fix flasky unittest for deformable psroi pooling

### PR DESCRIPTION
**Pull request by Oh233**
## Description ##
This PR aims to fix the problem raised by issue #11713. 

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
The deformable ps roi pooling operator includes several corner cases which the function is not differentiable. Therefore during the unittest, we need to add some check to ensure the input values are within the smooth regions.

## Comments ##
The issue and solution are discussed with @ankkhedia, @sandeep-krishnamurthy, @YuwenXiong and the fix is ran by @ankkhedia for about 2500 times on his local machine.

@ankkhedia could you help review it?

### CODE REVIEW

File: tests/python/unittest/test_operator.py [hunk 1]
```diff
@@ -5010,15 +5067,11 @@ def test_deformable_psroipooling():
...
-                        # wider tolerance needed for coordinate differential
-                        rtol, atol = 1.0, 1e-2
-                    else:
-                        rtol, atol = 1e-2, 1e-3
+                    rtol, atol = 1e-2, 1e-3
                     # By now we only have gpu implementation
                     if default_context().device_type == 'gpu':
                         check_numeric_gradient(op, [im_data, rois_data, offset_data], rtol=rtol, atol=atol,
-                                               grad_nodes=grad_nodes, ctx=mx.gpu(0))
+                                               grad_nodes=grad_nodes, ctx=mx.gpu(1))
```
Comment (marcoabreu) [inline tests/python/unittest/test_operator.py]: Is this intentional?

Comment (Oh233) [reply tests/python/unittest/test_operator.py]: Sorry, this is not intentional. I fixed it already.

**Comment by ankkhedia** [comment]
The changes looks good to me and have verified it locally for 2500 runs with random seeds.

**Comment by sandeep-krishnamurthy** [comment]
@marcoabreu - Your concerns are addressed. This is ready to be merged.

**Comment by ankkhedia** [comment]
@marcoabreu Your concerns are addressed. Could you please merge this PR?

**Comment by marcoabreu** [comment]
Everyone, please check  #11713

**Pull request merged**